lib/matplotlib/axes/_base.py

```diff
@@ -2875,6 +2875,11 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
         if right is not None:
             right = self.convert_xunits(right)
 
+        if ((left is not None and not np.isfinite(left)) or
+                (right is not None and not np.isfinite(right))):
+            raise ValueError("xlim limits must be finite. "
+                             "instead, found: (%s, %s)" % (left, right))
```

Comment:
Alright, sounds good. That redundancy bothered me too, but I couldn't think of the right verbiage to describe the warning. Thanks!